### ISSUE TITLE
Unexpected warning about a component being neither used nor exported

### ISSUE THREAD

**Issue by hunger** [slint-ui/slint#7176]
### Bug Description

I get a warning in my slint code that I do not expect:

```
Component is neither used nor exported [Ln 1, Col 1]
```

### Reproducible Code (if applicable)

```slint
component Foo /* Foo 1 */ { }

export component FooBaz {
    Foo /* <- TEST_ME_1 */ { }
}

component Foo /* Foo 2 */ { }

export component Baz {
    Foo /* <- TEST_ME_2 */ { }
}
```

### Environment Details

- Slint Version: master
- Platform/OS: Linux
- Programming Language: Not applicable
- Backend/Renderer: Not applicable

### Product Impact

Tiny :-)

**Comment by tronical** [comment]
I guess it should be an error that `Foo` is already defined?

**Comment by ogoffart** [comment]
Funny thing is that if we do NOT use Foo 1 (by commenting the line with TEST_ME_1), then we don't get that warning.
